File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.js
function solution(my_string, queries) {
  let strArray = my_string.split('');

  queries.forEach(([s, e]) => {
    let subArray = strArray.slice(s, e + 1).reverse();
    strArray.splice(s, e - s + 1, ...subArray);
  });

  return strArray.join('');
}

File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.java
import java.util.*;

class Solution {
    public String solution(String my_string, int[][] queries) {
        char[] answer = my_string.toCharArray();

        for (int[] query : queries) {
            int s = query[0];
            int e = query[1];
            
            while (s < e) {
                char temp = answer[s];
                answer[s] = answer[e];
                answer[e] = temp;
                s++;
                e--;
            }
        }
        
        return new String(answer);
    }
}